src/llvm-late-gc-lowering.cpp

```diff
@@ -728,8 +728,16 @@ State LateLowerGCFrame::LocalScan(Function &F) {
             Instruction &I = *it;
             if (CallInst *CI = dyn_cast<CallInst>(&I)) {
                 if (isa<IntrinsicInst>(CI)) {
-                    // Intrinsics are never GC uses/defs
-                    continue;
+                    // Most intrinsics are not gc uses/defs, however some have
+                    // memory operands and could thus be GC uses. To be conservative,
+                    // we only skip processing for those that we know we emit often
+                    // and cannot possibly be GC uses.
+                    IntrinsicInst *II = cast<IntrinsicInst>(CI);
+                    if (isa<DbgInfoIntrinsic>(CI) ||
+                        II->getIntrinsicID() == Intrinsic::lifetime_start ||
+                        II->getIntrinsicID() == Intrinsic::lifetime_end) {
+                        continue;
+                    }
```

Comment:
This version certainly looks better than the original one, which already looks good to me but you did remind me of a llvm intrinsic that I happened to notice the other day. [`llvm.load.relative`](http://llvm.org/docs/LangRef.html#llvm-load-relative-intrinsic) is the only instruction I've found so far that returns a derived pointer. Probably don't need to worry about it as long as LLVM optimization passes don't emit it?

Reply:
Thanks, I didn't know about that intrinsic (in my defensive it's only about a year old ;)), but yeah, I don't think we have to worry about it.